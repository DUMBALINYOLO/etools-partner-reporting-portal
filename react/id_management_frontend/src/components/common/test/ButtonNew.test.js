import React from 'react';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import ButtonNew from '../ButtonNew';

describe('ButtonNew component', () => {
    it('renders ButtonNew component and calls click', () => {
        const onClick = jest.fn();
        const classes = {};

        const wrapper = shallow(<ButtonNew
            onClick={onClick}
            classes={classes}
        />);

        wrapper.simulate('click');
        const calls = onClick.mock.calls;

        expect(calls.length).toBe(1);
        expect(toJSON(wrapper)).toMatchSnapshot();
    })
})