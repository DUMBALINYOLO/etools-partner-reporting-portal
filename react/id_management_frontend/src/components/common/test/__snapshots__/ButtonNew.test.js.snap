// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonNew component renders ButtonNew component and calls click 1`] = `
<ButtonNew
  classes={
    Object {
      "icon": "ButtonNew-icon-1",
    }
  }
  onClick={[Function]}
/>
`;
